frontend/sw360-portlet/src/main/webapp/html/projects/view.jsp

```diff
@@ -313,47 +313,81 @@
 		        });
 		    }
 
-		    function loadClearingStateSummaries() {
-		        var tableData = projectsTable.data();
-		        var ids = [];
-		        for (var i = 0; i < tableData.length; i++) {
-		            ids.push(tableData[i].id);
-		            var cell = projectsTable.cell(i, clearingSummaryColumnIndex);
-		            cell.data("Loading...");
-		        }
-		        jQuery.ajax({
-		            type: 'POST',
-		            url: '<%=loadClearingStateAjaxURL%>',
-		            cache: false,
-		            data: {
-		                "<portlet:namespace/><%=Project._Fields.ID%>": ids
-		            },
-		            success: function (response) {
-		                for (var i = 0; i < response.length; i++) {
-		                    var cell_clearingsummary = projectsTable.cell("#" + response[i].id, clearingSummaryColumnIndex);
-		                    cell_clearingsummary.data(displayClearingStateSummary(response[i].clearing));
-		                }
-		            },
-		            error: function () {
-		                for (var i = 0; i < tableData.length; i++) {
-		                    var cell = projectsTable.cell("#" + tableData[i].id, clearingSummaryColumnIndex);
-		                    cell.data("Failed to load");
-		                }
-		            }
-		        });
-		    }
+            function loadClearingStateSummaries() {
+                var tableData = projectsTable.data(),
+                    queryPageSize = 10,
+                    numberOfQueryPages = Math.ceil(tableData.length / queryPageSize),
+                    queryPagesPageIds = [],
+                    pageIds,
+                    cell,
+                    i, j, idx;
+
+                for (i = 1; i <= numberOfQueryPages; i++) {
+                    pageIds = [];
+                    for (j = 0; j < queryPageSize; j++) {
+                        idx = ((i-1) * queryPageSize) + j;
```

Comment:
this is really a minor thing, but you have the `i`-loop from 1 till `<= numberOfQueryPages` and the at only place you use the `i ` you decrement it by 1. why not doing it the conventional way from 0 to `< numberOfQueryPages` ?

Reply:
Haha, I thought that was the idea on using Math.ceil so that the loop loops over the visible page numbers :D But now I got you: It's all about the variable value being correct according to its name.